netdumpd.c

```diff
@@ -388,6 +396,26 @@ alloc_client(int sd, struct sockaddr_in *saddr, char *path)
 		if (firstdot)
 			*firstdot = '\0';
 	}
+#else
+	error = getnameinfo((struct sockaddr *)saddr,
+	    saddr->sin_len, client->hostname, sizeof(client->hostname),
+	    NULL, 0, NI_NAMEREQD);
+	if (error != 0) {
+		/* Can't resolve, try with a numeric IP. */
+		error = getnameinfo((struct sockaddr *)saddr,
+		    saddr->sin_len, client->hostname, sizeof(client->hostname),
+		    NULL, 0, 0);
+		if (error != 0) {
+			LOGERR("getnameinfo(): %s\n", gai_strerror(error));
+			goto error_out;
+		}
+	} else {
+		/* Strip off the domain name. */
+		firstdot = strchr(client->hostname, '.');
+		if (firstdot)
+			*firstdot = '\0';
+	}
+#endif
```

Comment:
You should be able to do something like

```
#define cap_getnameinfo(channel, ...) getnameinfo(__VA_ARGS__)
```

in the non-capsicum case, and then everything ought to work without any code duplication.